src/Interpreters/join_common.cpp

```diff
@@ -283,6 +285,87 @@ void addDefaultValues(IColumn & column, const DataTypePtr & type, size_t count)
         type->insertDefaultInto(column);
 }
 
+bool typesEqualUpToNullability(DataTypePtr left_type, DataTypePtr right_type)
+{
+    DataTypePtr left_type_strict = removeNullable(recursiveRemoveLowCardinality(left_type));
+    DataTypePtr right_type_strict = removeNullable(recursiveRemoveLowCardinality(right_type));
+    return left_type_strict->equals(*right_type_strict);
+}
+
+JoinConvertActions columnsNeedConvert(const Block & left_block, const Names & left_keys,
+                                      const Block & right_block, const Names & right_keys,
+                                      bool has_using)
+{
+    assert(left_keys.size() == right_keys.size());
+
+    /// only JOIN USING supported
```

Comment:
The fact that only `USING` is supported is clear enough from the code, the comment should probably tell "why".

Reply:
I've added comment to TreeRewriter, where we start to deal with join column conversion and to InterpreterSelectQuery where step added to query plan.
https://github.com/ClickHouse/ClickHouse/pull/19885/files#diff-03665017349505c5529e5d0eeeeae80081e064267d6035b94cd428baf11856ceR417
https://github.com/ClickHouse/ClickHouse/pull/19885/files#diff-5452971ae18972aa669265008a29b3732a99d9415f5b3af10d573c0807667a22R978